KB: Expand/Collapse Detail page on RowClick or DoubleRowClick (#746)
telerik/blazor-docs: knowledge-base/grid-expand-collapse-detail-onclick-or-ondoubleclick.md

---
title: Expand/Collapse Detail page on RowClick or DoubleRowClick
description: How to expand/collapse Detail page on RowClick or DoubleRowClick instead of clicking the expand/collapse button?
type: how-to
page_title: Expand/Collapse Detail page on RowClick or DoubleRowClick
slug: grid-kb-expand-collapse-detail-onclick-or-ondoubleclick
position: 
tags: grid, hierarchy, detail, expand, collapse, onclick, ondoubleclick
ticketid: 1548965
res_type: kb
---

## Environment
<table>
	<tbody>
		<tr>
			<td>Product</td>
			<td>Grid for Blazor</td>
		</tr>
	</tbody>
</table>

## Description

I want to expand and collapse the Grid Details on row click or double click instead of clicking the expand/collapse button.

## Solution

> You can handle the Grid events ([`OnRowClick`]({%slug grid-events%}#onrowclick) or [`OnRowDoubleClick`]({%slug grid-events%}#onrowdoubleclick)) to programmatically expand/collapse the `DetailTemplate` of an item.

Reviewer: ```suggestion
Handle the Grid events ([`OnRowClick`]({%slug grid-events%}#onrowclick) or [`OnRowDoubleClick`]({%slug grid-events%}#onrowdoubleclick)) to programmatically expand/collapse the `DetailTemplate` of an master item.
```

                <GridColumns>
                    <GridColumn Field="OrderId"></GridColumn>
                    <GridColumn Field="DealSize"></GridColumn>
                </GridColumns>
            </TelerikGrid>
        }
    </DetailTemplate>
    <GridColumns>
        <GridColumn Field="Id"></GridColumn>
        <GridColumn Field="Name"></GridColumn>
    </GridColumns>
</TelerikGrid>

@code {
    List<MainModel> salesTeamMembers { get; set; }
    TelerikGrid<MainModel> Grid { get; set; }

    async Task OnRowDoubleClickHandler(GridRowClickEventArgs args)
    {
        var currItem = args.Item as MainModel;

        var state = Grid.GetState();

        bool isCurrItemExpanded = state.ExpandedItems.Any(x => x.Id == currItem.Id);

        if (isCurrItemExpanded)
        {
            state.ExpandedItems.Remove(currItem);
        }else{
            state.ExpandedItems.Add(currItem);
        }

        await Grid.SetState(state);
    }

    protected override void OnInitialized()
    {
        salesTeamMembers = GenerateData();
    }

    private List<MainModel> GenerateData()
    {
        List<MainModel> data = new List<MainModel>();

>         for (int i = 0; i < 5; i++)

Reviewer: ```suggestion
        for (int i = 1; i <= 5; i++)
```
